components/net/indexeddb/engines/sqlite.rs

```
@@ -0,0 +1,513 @@
+                .filter(object_store_model::Column::Name.eq(store_name.to_string()))
+                .one(&self.connection)
+                .await?
+            {
+                object_store_index_model::Entity::delete_many()
+                    .filter(object_store_index_model::Column::ObjectStoreId.eq(store.id))
+                    .exec(&self.connection)
+                    .await?;
+                object_data_model::Entity::delete_many()
+                    .filter(object_data_model::Column::ObjectStoreId.eq(store.id))
+                    .exec(&self.connection)
+                    .await?;
+                store.delete(&self.connection).await?;
+            }
+            Ok(())
+        })
+    }
+
+    fn close_store(&self, _store_name: SanitizedName) -> Result<(), Self::Error> {
+        HANDLE.block_on(async {
+            // TODO: do something
+            Ok(())
+        })
+    }
+
+    fn delete_database(self) -> Result<(), Self::Error> {
+        HANDLE.block_on(async {
+            self.connection.close().await?;
+            if self.db_path.exists() {
+                std::fs::remove_dir_all(self.db_path.parent().unwrap()).unwrap();
+            }
+            Ok(())
+        })
+    }
+
+    fn process_transaction(
+        &self,
+        transaction: KvsTransaction,
+    ) -> oneshot::Receiver<Option<Vec<u8>>> {
+        let (tx, rx) = oneshot::channel();
+        let connection = self.connection.clone();
+
+        // TODO: maybe use different pools for different transactions?
+        HANDLE.spawn(async move {
+            for request in transaction.requests {
+                let conn = connection.clone();
+                let object_store = match object_store_model::Entity::find()
+                    .filter(object_store_model::Column::Name.eq(request.store_name.to_string()))
+                    .one(&conn)
+                    .await
+                    .unwrap()
```

What are we unwrapping here? Can we return an error instead?